src/Compiler/Optimize/Optimizer.fs

```diff
@@ -4081,6 +4099,9 @@ and OptimizeBinding cenv isRec env (TBind(vref, expr, spBind)) =
                     if fvs.UsesMethodLocalConstructs then
                         // Discarding lambda for binding because uses protected members
                         UnknownValue 
+                    elif fvs.FreeLocals.ToArray() |> Seq.fold(fun acc v -> if not acc then v.Accessibility.IsPrivate else acc) false then
```

Comment:
```suggestion
                    elif fvs.FreeLocals.Exists(fun v -> v.Accessibility.IsPrivate) false then
```

Reply:
Added this comment:
       // Previous versions of the compiler used the inline flag to signal that inline must happen.
       // With the introduction of realsig, we have relaxed that requirement.
       // We now offer a warning as a signal to  rework the expression when accessing unreachable values.
       // We do this because private values in realsig- are emitted as 'assembly' which can be accessed
       // in in-assembly and cross-assembly inlining scenarios with friend assemblies.
       // But with realsig+ those values are emitted 'private' which will not work for most all
       // inline scenarios.  In order to improve the compatability of realsig+ code with existing code,
       // we relax the meaning of inline to should inline.  Similar to how we treat LambdaInline.